Windows.md

```diff
@@ -17,29 +17,70 @@
+may install additional features from Online sources in the Online menu.
 
-Once you have CMAKE support available please select the appropriate target. A file named CMakeSettings.json provides a few base configurations, x64-RelWithDebInfo, x64-Release,
-x64-RelWithDebInfo-WithJNI, and x64-Release-WithJNI. Though the project is written in C++, JNI functionality supports running Java processors stored in NiFi Archives. These can be run
-in a much smaller memory footprint and consumer fewer resources. If your systems do not support Java or you do not want a JDK installed, please select the non JNI targets.
+A file named CMakeSettings.json provides the CMake configuration.
 
-CMAKE must generate its cache, under Cache in the CMake Menu. After that is complete go to 'Build Only' under the CMAKE menu. Due to limitations in Visual Studio's CMAKE support, it is advised
-that you build minifi.lib then minifiexe targets. Once you have built these targets, you may use the cpack command to build your MSI. If you are building with JNI functionality the MSI will be
+CMake must generate its cache, under Cache in the CMake Menu. After that is complete go to 'Build Only' under the CMake menu. Due to limitations in Visual Studio's CMake support, it is advised
+that you build `minifi.lib` then `minifiexe` targets. Once you have built these targets, you may use the `cpack` command to build your MSI. If you are building with JNI functionality the MSI will be
 significantly larger ( about 160 MB ) since it contains the base NARs to run the standard set of Apache NiFi processors. 
 
 ## Building via the build script
 
-The preferred way of building the project is via the win_build_vs.bat script found in our root source folder. You must supply a single command, the build directory. Typically you create a directory
-with CMAKE and build in that referencing your CMAKE tree. Simply supply win_build_vs.bat an argument like 'build' as the name of your build directory and it will create this directory building the 
-project within it. If WiX is installed cpack will create your MSI in the chosen build directory.
+The preferred way of building the project is via the `win_build_vs.bat` script found in our root source folder. Its first parameter is mandatory, the directory in which it will build the project. `build` is a good default choice for this.
+
+After the build directory it will take optional parameters modifying the CMake configuration used in the build:
+
+| Argument | Effect |
+|----------|------------------------------------------------|
+| /T | Disables tests |
+| /P | Enables MSI creation |
+| /K | Enables Kafka extension |
+| /J | Enables JNI |
+| /64 | Creates 64-bit build instead of a 32-bit one |
+| /D | Builds RelWithDebInfo build instead of Release |
+
+Examples:
+ - 32-bit build with kafka, disabling tests, enabling MSI creation: `win_build_vs.bat build /T /K /P`
+ - 64-bit build with JNI, with debug symbols: `win_build_vs.bat build /64 /J /D`
```

Comment:
I think our docs shouldn't (even implicitly) recommend building 32 and 64 bit binaries to the same there, so I would prefer changing the 1st arg to "build64" here.

Reply:
I didn't want to contradict myself by saying that build is a good default choice, but yeah, different names would be safer, I'll change it.